consensus/src/payload_manager.rs

```diff
@@ -94,26 +94,36 @@ impl PayloadManager for QuorumStoreClient {
             Err(anyhow::anyhow!("Injected error in pull_payload").into())
         });
         let mut callback_wrapper = Some(wait_callback);
-        // keep polling QuorumStore until there's payloads available or there's still pending payloads
+        // keep polling QuorumStore until we can make large enough payload, from available or still pending payloads
         let mut count = self.poll_count;
-        let payload = loop {
+        loop {
             count -= 1;
             let payload = self
                 .pull_internal(max_items, max_bytes, exclude_payloads.clone())
                 .await?;
-            if payload.is_empty() && !pending_ordering && count > 0 {
+
+            if (payload.len() as u64) < max_items
+                && payload.bytes() < max_bytes
```

Comment:
I think this will unnecessarily wait in the case we hit max_bytes, but not exactly.

Example: max_items = 10, max_bytes = 7 bytes
txn1, txn2, txn3 = 3 bytes

The returned payload will be [txn1, txn2] and bytes = 6. so we'll ask again.